src/global_scheduler/global_scheduler.c

```diff
@@ -41,6 +51,14 @@ void assign_task_to_local_scheduler(global_scheduler_state *state,
       get_local_scheduler(state, local_scheduler_id);
   local_scheduler->num_tasks_sent += 1;
   local_scheduler->num_recent_tasks_sent += 1;
+  /** Resource accounting update for this local scheduler. */
+  for (int i = 0; i < MAX_RESOURCE_INDEX; i++) {
+    /** Subtract task's resource from the cached dynamic resource capacity for
+     *  this local scheduler. This will be overwritten on the next heartbeat. */
+    local_scheduler->info.dynamic_resources[i] =
+        MAX(0, local_scheduler->info.dynamic_resources[i] -
```

Comment:
I'm worried this may have unexpected effects on the score. Just letting it be negative doesn't have the effect you'd want on the score. E.g. dyn_cpu_node1 == -2, dyn_cpu_node2 = -1 and requested cpu=1. Node1 would cause less negative delta in the score than node2, and we'd want it to be reversed. dynfrac_cpu_node1 = -0.5 and dynfrac_cpu_node2 = -1. So node2 appears worse, while it should be better.